proposals/gc/MVP.md

```diff
@@ -273,62 +300,112 @@ Tentatively, support a type of guaranteed unboxed scalars.
 
 * `i31.new` creates an `i31ref` from a 32 bit value, truncating high bit
   - `i31.new : [i32] -> [i31ref]`
+  - this is a *constant instruction*
 
 * `i31.get_u` extracts the value, zero-extending
   - `i31.get_u : [i31ref] -> [i32]`
 
 * `i31.get_s` extracts the value, sign-extending
   - `i31.get_s : [i31ref] -> [i32]`
 
-Perhaps also the following short-hands:
+
+#### Classification
```

Comment:
Is it intentional that there are no instructions for eqref, i.e. `ref.is_eq`/`ref.as_eq`/`br_on_eq`? Previously it was possible to cast from anyref ot eqref, though I'm not sure how relevant that is in practice. In particular, if we end up making i31 an attribute on references (like nullability), then `ref.is_data` will effectively subsume `ref.is_eq`.

Reply:
Yes. Eqref isn't really a basic category of values but rather a constraint on anyref. I can't think of a good use case to cast to it, and it doesn't seem like that would be a simple operation to implement either.